### PR TITLE
ADD Class 9

### PR DESCRIPTION
**Pull request by EvanSimpson**
Application Design and Diagramming

### CODE REVIEW

File: classes/class09/README.md [hunk 1]
```diff
@@ -1 +1,129 @@
...
+|  +--login.handlebars
++--models/
+|  +--users.js
+|  +--twotes.js
++--routes/
+|  +--users.js
+|  +--twotes.js
++--util/
++--app.js
++--package.json
```
Comment (benkahle) [inline classes/class09/README.md:136]: tests?

File: classes/class09/README.md [hunk 2]
```diff
@@ -1 +1,129 @@
...
+```
+
+##Exercise
+Form groups of 4-5 people and select one of the following websites.
+We're going to deconstruct these websites and design a simple specification for these sites as if we were going to implement them ourselves.
+Don't worry about trying to capture all of the functionality of these sites and instead focus on a few core features that you might reasonably be able to implement at this time.
+Expect to spend about 10-15 minutes on this exercise.
```
Comment (benkahle) [inline classes/class09/README.md]: I was thinking more like 25-30 min

File: classes/class09/README.md [hunk 3]
```diff
@@ -1 +1,129 @@
...
++--routes/
+|  +--users.js
+|  +--twotes.js
++--util/
++--app.js
++--package.json
+```
+
+##Exercise
+Form groups of 4-5 people and select one of the following websites.
```
Comment (benkahle) [inline classes/class09/README.md:166]: We might specify (or maybe just in class) that each group should have a different site.

File: classes/class09/README.md [hunk 4]
```diff
@@ -1 +1,129 @@
...
+ - Other requirements
+
+There is no hard and fast rule about how long a spec should be, but including all of that is just plain unnecessary for anything we'll be making in this class.
+Instead, let's distill some of the key ideas from above that our specs should touch on.
+
+***Disclaimer:*** Your teachers are by no means experts in writing specifications.
+The suggestions offered below are just that.
+The correct format will vary by project, so try to find what works best for you.
+
+- **Overview:** The story of your application, justification for why it needs to exist, and a broad overview of what it will accomplish.
```
Comment (benkahle) [inline classes/class09/README.md:55]: Were you planning to add examples of these, they are still pretty vague for most people I think.

File: classes/class09/README.md (hunk 1 above)
Comment (EvanSimpson) [inline classes/class09/README.md:136]: :+1:

File: classes/class09/README.md (hunk 4 above)
Comment (EvanSimpson) [inline classes/class09/README.md:55]: I had no plans. But we can figure something out.

:game_die: :game_die:

File: classes/class09/README.md (hunk 3 above)
Comment (EvanSimpson) [inline classes/class09/README.md:166]: I think that's fine as an in-class thing.

**Comment by EvanSimpson** [comment]
@benkahle is that in reference to a specific line?

**Comment by benkahle** [comment]
Yeah, click on the commit hash to jump to it.

**Comment by EvanSimpson** [comment]
Got it, thank you.

File: classes/class09/README.md (hunk 4 above)
Comment (EvanSimpson) [inline classes/class09/README.md:55]: We can go through an example in class. We'll have time.

File: classes/class09/README.md [hunk 5]
```diff
@@ -1 +1,179 @@
...
+If you ever get to the point that you're working on a *really* big application that's distributed across multiple different severs, you'll need to start diagramming a map of all of your different servers, databases, and other networked components, and how they all connect.
+We'll never get around to anything remotely close to the system shown below (in fact I'd be surprised if anyone has a configuration like that), but the principle is still there.
+![AWS network layout](images/AWSExample.png)
+
+##Refactoring
+As you develop (or let's be honest - have been developing), you might find yourself doing hacky things just to get your code to work.
+Avoid this if you can.
+However if you do, it's valuable to take a few minutes to **refactor** your code.
+If you're unfamiliar with this term, refactoring basically refers to restructuring how your code is laid out across files, functions, modules, etc. without necessarily modifying any functionality.
+In order for you to be effective at refactoring, let's go through a typical folder structure for one of our applications and look at what goes where.
```
Comment (benkahle) [inline classes/class09/README.md:107]: Its probably worth adding a bit more about refactoring code in terms of how functions/objects/classes interact. The current description makes it sound like its mostly just what files and modules code chunks are placed in.

Comment (EvanSimpson) [inline classes/class09/README.md:107]: Will add at future date.

**Comment by EvanDorsky** [comment]
We taught this. :+1:

**Pull request merged**